src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -514,37 +512,31 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
         case QgsUnitTypes::TemporalIrregularStep:
           return QString();
       }
-      return QStringLiteral( "(%1 %2 %3 OR %1 IS NULL) AND ((%1 + %4 %5 %6) OR %7 IS NULL)" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-             range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
+      return QStringLiteral( "(%1 < %2 OR %1 IS NULL) AND ((%1 + %3 > %4) OR %5 IS NULL)" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
              dateTimeExpressionLiteral( range.end() ),
              intervalExpression,
-             range.includeBeginning() ? QStringLiteral( ">=" ) : QStringLiteral( ">" ),
              dateTimeExpressionLiteral( range.begin() ),
              QgsExpression::quotedColumnRef( mDurationFieldName ) );
-      break;
     }
 
     case ModeFeatureDateTimeStartAndEndFromFields:
     {
       if ( !mStartFieldName.isEmpty() && !mEndFieldName.isEmpty() )
       {
-        return QStringLiteral( "(%1 %2 %3 OR %1 IS NULL) AND (%4 %5 %6 OR %4 IS NULL)" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
+        return QStringLiteral( "(%1 < %2 OR %1 IS NULL) AND (%3 > %4 OR %3 IS NULL)" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
                dateTimeExpressionLiteral( range.end() ),
                QgsExpression::quotedColumnRef( mEndFieldName ),
-               range.includeBeginning() ? QStringLiteral( ">=" ) : QStringLiteral( ">" ),
                dateTimeExpressionLiteral( range.begin() ) );
       }
       else if ( !mStartFieldName.isEmpty() )
       {
-        return QStringLiteral( "%1 %2 %3 OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
-               range.includeBeginning() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
+        return QStringLiteral( "%1 < %2 OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
```

Comment:
I think the logic here is broken (completely in the current code!)

I believe the correct one is:

```
return QStringLiteral( "%1 %2 %3 OR %1 IS NULL" ).arg( QgsExpression::quotedColumnRef( mStartFieldName ),
               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
```